cfme/middleware/provider/__init__.py

```diff
@@ -82,6 +87,80 @@ class MiddlewareProvider(BaseProvider):
     add_provider_button = form_buttons.FormButton("Add this Middleware Provider")
     save_button = form_buttons.FormButton("Save Changes")
     taggable_type = 'ExtManagementSystem'
+    appliance = CurrentAppliance()
+
+
+@navigator.register(MiddlewareProvider, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Middleware', 'Providers')(None)
+
+    def resetter(self):
+        # Reset view and selection
+        tb.select("Grid View")
+        sel.check(paginator.check_all())
+        sel.uncheck(paginator.check_all())
+
+
+@navigator.register(MiddlewareProvider, 'Add')
+class Add(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        cfg_btn('Add a New Middleware Provider')
+
+
+@navigator.register(MiddlewareProvider, 'Details')
+class Details(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        sel.click(Quadicon(self.obj.name, self.obj.quad_name))
+
+
+@navigator.register(MiddlewareProvider, 'Edit')
+class Edit(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        sel.check(Quadicon(self.obj.name, self.obj.quad_name).checkbox())
+        cfg_btn('Edit Selected Middleware Provider')
+
+
+@navigator.register(MiddlewareProvider, 'EditFromDetails')
+class EditFromDetails(CFMENavigateStep):
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self):
+        cfg_btn('Edit this Middleware Provider')
+
+
+@navigator.register(MiddlewareProvider, 'EditTags')
+class EditTags(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        sel.check(Quadicon(self.obj.name, self.obj.quad_name).checkbox())
+        pol_btn('Edit Tags')
+
+
+@navigator.register(MiddlewareProvider, 'EditTagsFromDetails')
+class EditTagsFromDetails(CFMENavigateStep):
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self):
+        pol_btn('Edit Tags')
+
+
+@navigator.register(MiddlewareProvider, 'TimelinesFromDetails')
```

Comment:
In cloud provider I just called this destination 'Timelines', since there was only one way to navigate there (from details).

That being said, I just want to be consistent, I don't have strong feelings about doing it either way, and there is definitely some value to the path being described by the destination name. Runs the risk of ending up with a name that's too long, but we can address that as we find it.